### PR TITLE
Split settings page components into files, enhance loading speed

### LINKED ISSUE #2353: Remove one of the three "More Settings" buttons from popup
**Is your feature request related to a problem? Please describe.**

When the "Addons" tab is selected in the popup, there are now three buttons that open the settings page (including the one in the popup header).

**Describe the solution you'd like**

Remove the More Settings button next to the search bar.

**Describe alternatives you've considered**

Remove the newly added one instead and change the message above it to say "Click *More settings* above to explore all 80+ addons".

### PR DESCRIPTION
**Pull request by WorldLanguages**
Related: #2531
- Create an HTML and JS file for every component in webpages/settings/components, and import them using the load-vue-components library (#2424) Files were moved almost as-is, with no significant changes. Commits: https://github.com/ScratchAddons/ScratchAddons/pull/2567/commits/bee85aac367fee3cbd542d41a1a9a6463cbb92e8
- Create JS files for static data, like tag array, addon groups array and addon categories, in webpages/settings/data. Commits: https://github.com/ScratchAddons/ScratchAddons/pull/2567/commits/7c36c5f6b7f7ad2639d126702a698c3c7d011134 https://github.com/ScratchAddons/ScratchAddons/pull/2567/commits/a8a373272434518f2cc8179bb9234b9ac42a8bac
- Use `<link rel="preload">` in head to preload index.js module imports as well as component import/fetches. Remove implicit "defer" in module script. Commits: https://github.com/ScratchAddons/ScratchAddons/commit/bd77003987e0d5849e9aa94f1479e821ed61460f
- Reduce layout shift by using `<link rel="preload">` for Sora font, as well as setting both the width and height for square images. Commits: https://github.com/ScratchAddons/ScratchAddons/commit/a7601f246f0ccc0ec11133bb8654c89517946c5a
- Set both the width and height for more square images. Set default dir to ltr (no reason in particular really, for the same reason we have lang="en" by default. we should remove both or keep both). Preload images using  `<link rel="preload">`. Show body element as soon as we know whether to use dark or light mode (without data being ready) only in fullscreen mode (popup mode works the same as before - body element is hidden until both data and Vue are ready). Hide `{{ msg("string") }}` while loading. Load theme data ASAP, without waiting to load components. Avoid transition on `.category` elements while strings are loading (edit: later removed for being unnecessary, probably happened before because of a layout shift fix I made later). Commits: https://github.com/ScratchAddons/ScratchAddons/commit/5302d8925e9d4146a8fb543f9110e1f8bf58a490
- For faster load, do not create `<addon-body>` elements for search until 350ms have passed since the user typed in the seach bar and then stopped typing for that period of time. https://github.com/ScratchAddons/ScratchAddons/commit/cc2aeacba3b637bd6b1f156f608c11e2c28a9df5
- Resolves #2353 Commits: https://github.com/ScratchAddons/ScratchAddons/pull/2567/commits/58cc2796ce02c49ed03d4ac0e88f0cf494622d41

### CODE REVIEW

**Comment by apple502j** [comment]
I hope this doesn't break in FF84?

**Comment by WorldLanguages** [comment]
> I hope this doesn't break in FF84?

No, preload is simply an optimization. Not required

**Comment by WorldLanguages** [comment]
@GrahamSH-LLK Can you confirm if this line of CSS can be removed safely? It doesn't seem to do anything, and it causes layout shift. https://github.com/ScratchAddons/ScratchAddons/pull/2567/files#diff-267b5d7d5d1e6e17e73cf3070e675ff261a6b238bbae8dce60dbcdfc66830e39R124

**Comment by mxmou** [comment]
> @GrahamSH-LLK Can you confirm if this line of CSS can be removed safely? It doesn't seem to do anything, and it causes layout shift. https://github.com/ScratchAddons/ScratchAddons/pull/2567/files#diff-267b5d7d5d1e6e17e73cf3070e675ff261a6b238bbae8dce60dbcdfc66830e39R124

It prevents addon titles from wrapping on small window sizes.

**Comment by WorldLanguages** [comment]
@mxmou I don't think I understood what it does. Keep in mind the style is applied to the img element in infos and warns.

**Comment by WorldLanguages** [comment]
Merging for v1.15.0 - waiting for confirmation on how to keep `flex: 0 0` while still removing layout shift

**Comment by TheColaber** [comment]
TODO: Move some css files from `webpages/styles/components` to `webpages/settings/components`?

**Comment by mxmou** [comment]
> @mxmou I don't think I understood what it does. Keep in mind the style is applied to the img element in infos and warns.

I read the diff wrong. I think it would work as expected (prevent the image from shrinking) if the notice itself would be in a `<span>` to act as a flex item, but with the current template the line is unnecessary.

File: webpages/settings/index.html [hunk 1]
```diff
@@ -31,287 +104,22 @@
...
+        :src="switchPath"
+        class="toggle"
+        @click="sidebarToggle()"
+        v-show="smallMode"
+        alt="Logo"
+      />
       <img src="../../images/icon.svg" class="logo" alt="Logo" />
-      <h1>{{ msg("settings") }}</h1>
-      <img @click="setTheme(!theme)" class="theme-switch" v-bind:src="themePath" />
+      <h1 style="display: none" v-show="true">{{ msg("settings") }}</h1>
```
Comment (GrahamSH-LLK) [inline webpages/settings/index.html:121]: v-cloak, people

Comment (retronbv) [reply webpages/settings/index.html:121]: I’m dumb but what does v-cloak do

Comment (GrahamSH-LLK) [reply webpages/settings/index.html:121]: Vue automatically removes v-cloak when it loads. If you put a css rule to hide v-cloak items, you don't need v-show or display:none multiple times.

**Comment by WorldLanguages** [comment]
> TODO: Move some css files from `webpages/styles/components` to `webpages/settings/components`?

I don't think we should touch CSS yet.

File: webpages/settings/index.html (hunk 1 above)
Comment (WorldLanguages) [reply webpages/settings/index.html:121]: @GrahamSH-LLK But then the elements are visible until Vue loads

Comment (WorldLanguages) [reply webpages/settings/index.html:121]: Could be fixed with
```css
[v-cloak] {
  display: none;
}
```

Comment (RedGuy12) [reply webpages/settings/index.html]: Doesn't the body have display none?

Comment (GrahamSH-LLK) [reply webpages/settings/index.html]: > @GrahamSH-LLK But then the elements are visible until Vue loads

> If you put a css rule to hide v-cloak items

File: webpages/settings/index.html [hunk 2]
```diff
@@ -1,14 +1,87 @@
...
+      as="font"
+      type="font/woff2"
+      crossorigin
+    />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <title></title>
     <link rel="icon" href="../../images/icon.png" />
+
+    <!-- Preload index.js imports -->
+    <link rel="modulepreload" href="../../libraries/common/cs/download-blob.js"">
```
Comment (mxmou) [inline webpages/settings/index.html:45]: Typo (should end with `download-blob.js" />`, not `download-blob.js"">`)

Comment (TheColaber) [reply webpages/settings/index.html:45]: Yeah, this has been very annoying in my atom editor, but my vscode editor doesn't care. Prettier doesn't care either.

**Pull request merged**